src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -632,7 +685,7 @@ private void ValidateSolutionFileVersion(string versionString)
         ///  EndProject
         /// 
         /// </summary>
-        private void ParseProject(string firstLine)
+        private void ParseProject(string firstLine, HashSet<string> slnFilter, HashSet<string> guidFilter)
```

Comment:
What is `guidFilter` for?

Reply:
It's for nested projects; they refer to each other by guid instead of path, so I needed a second way of determining if a project should be included or not.